nodejs: fix cross-compilation from aarch64-linux to x86_64-linux

tie opened the pull request:
###### Description of changes

This PR adds patches for Node.js that fix cross-compilation for v18 and v20. It also enables checkPhase for a subset of test suite that does not require dependencies not bundled with the tarball.

See also https://github.com/NixOS/nixpkgs/pull/234124#issuecomment-1594211731

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

tie committed:
nodejs_{18,20}: fix cross compilation from aarch64-linux

This change adds a fix for Node.js build system passing flags intended
for the target toolchain to the host toolchain.

tie committed:
nodejs: do not pass --without-snapshot

Building --without-snapshot is no longer possible since v14. See
https://github.com/nodejs/node/commit/0822bfaa9f52aea289abd0e15a90b0df2e12f7e1

tie committed:
nodejs_18: backport v8 trap handler conditional compilation

tie committed:
nodejs: do not run patchShebangs with build inputs

tie committed:
nodejs: install shell completions and man pages when cross-compiling

tie committed:
nodejs: mark as broken when cross-compiling for different OS

tie committed:
nodejs: run a subset of test suite

tie committed:
nodejs: patch shebangs for shell scripts when cross-compiling

tie committed:
nodejs: remove no-op postPatch phase

tr1/type_traits replacement is only relevent for Node.js v6 that’s
reached EOL in 2019.

tie committed:
nodejs: cross-compile with icu

tie commented:
Result of `nixpkgs-review pr --checkout commit 238377` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>77 packages marked as broken and skipped:</summary>
  <ul>
    <li>calligra</li>
    <li>code-server</li>
    <li>coder</li>
    <li>discourse</li>
    <li>discourseAllPlugins</li>
    <li>github-runner</li>
    <li>kexi</li>
    <li>kibana</li>
    <li>kibana7</li>
    <li>libsForQt5.full</li>
    <li>libsForQt5.kreport</li>
    <li>libsForQt5.qt5.full</li>
    <li>magnetophonDSP.ConstantDetuneChorus</li>
    <li>mermaid-filter</li>
    <li>minizincide</li>
    <li>mx-puppet-discord</li>
    <li>nodejs-slim_14</li>
    <li>nodejs-slim_14.libv8</li>
    <li>nodejs-slim_16</li>
    <li>nodejs-slim_16.libv8</li>
    <li>nodejs_14</li>
    <li>nodejs_14.libv8</li>
    <li>nodejs_16</li>
    <li>nodejs_16.libv8</li>
    <li>npm-check</li>
    <li>odoo</li>
    <li>odoo.dist</li>
    <li>oni2</li>
    <li>openvscode-server</li>
    <li>plasma5Packages.full</li>
    <li>plasma5Packages.kreport</li>
    <li>plasma5Packages.qt5.full</li>
    <li>python310Packages.batchspawner</li>
    <li>python310Packages.batchspawner.dist</li>
    <li>python310Packages.caffeWithCuda</li>
    <li>python310Packages.caffeWithCuda.bin</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.dockerspawner</li>
    <li>python310Packages.dockerspawner.dist</li>
    <li>python310Packages.jupyterhub</li>
    <li>python310Packages.jupyterhub-ldapauthenticator</li>
    <li>python310Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python310Packages.jupyterhub-systemdspawner</li>
    <li>python310Packages.jupyterhub-systemdspawner.dist</li>
    <li>python310Packages.jupyterhub-tmpauthenticator</li>
    <li>python310Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python310Packages.jupyterhub.dist</li>
    <li>python310Packages.oauthenticator</li>
    <li>python310Packages.oauthenticator.dist</li>
    <li>python310Packages.optuna</li>
    <li>python310Packages.optuna.dist</li>
    <li>python311Packages.batchspawner</li>
    <li>python311Packages.batchspawner.dist</li>
    <li>python311Packages.dockerspawner</li>
    <li>python311Packages.dockerspawner.dist</li>
    <li>python311Packages.jupyterhub</li>
    <li>python311Packages.jupyterhub-ldapauthenticator</li>
    <li>python311Packages.jupyterhub-ldapauthenticator.dist</li>
    <li>python311Packages.jupyterhub-systemdspawner</li>
    <li>python311Packages.jupyterhub-systemdspawner.dist</li>
    <li>python311Packages.jupyterhub-tmpauthenticator</li>
    <li>python311Packages.jupyterhub-tmpauthenticator.dist</li>
    <li>python311Packages.jupyterhub.dist</li>
    <li>python311Packages.oauthenticator</li>
    <li>python311Packages.oauthenticator.dist</li>
    <li>qt5.full</li>
    <li>qt5Full</li>
    <li>teleprompter</li>
    <li>trojita</li>
    <li>tumpa</li>
    <li>tumpa.dist</li>
    <li>vimPlugins.coc-imselect</li>
    <li>whalebird</li>
    <li>zerobin</li>
    <li>zerobin.dist</li>
    <li>zombietrackergps</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.writers</li>
  </ul>
</details>
<details>
  <summary>38 packages failed to build:</summary>
  <ul>
    <li>botamusique</li>
    <li>bs-platform</li>
    <li>caffeWithCuda</li>
    <li>caffeWithCuda.bin</li>
    <li>geph.gui</li>
    <li>gpu-screen-recorder</li>
    <li>gpu-screen-recorder-gtk</li>
    <li>jiten</li>
    <li>jiten.dist</li>
    <li>openai-full</li>
    <li>openai-full.dist</li>
    <li>pulumiPackages.pulumi-language-nodejs</li>
    <li>python310Packages.fastai</li>
    <li>python310Packages.fastai.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.langchain</li>
    <li>python310Packages.langchain.dist</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.skrl.dist</li>
    <li>python310Packages.spacy</li>
    <li>python310Packages.spacy-loggers</li>
    <li>python310Packages.spacy-loggers.dist</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.spacy-transformers.dist</li>
    <li>python310Packages.spacy.dist</li>
    <li>python310Packages.textacy</li>
    <li>python310Packages.textacy.dist</li>
    <li>python310Packages.textnets</li>
    <li>python310Packages.textnets.dist</li>
    <li>python310Packages.wandb</li>
    <li>python310Packages.wandb.dist</li>
    <li>python311Packages.caffeWithCuda</li>
    <li>python311Packages.caffeWithCuda.bin</li>
    <li>replay-node-cli</li>
    <li>sitespeed-io</li>
    <li>sonic-pi</li>
    <li>timetagger</li>
  </ul>
</details>
<details>
  <summary>829 packages built:</summary>
  <ul>
    <li>CuboCore.corepdf</li>
    <li>PageEdit</li>
    <li>activitywatch</li>
    <li>adl</li>
    <li>adoptopenjdk-icedtea-web</li>
    <li>akkoma-frontends.admin-fe</li>
    <li>akkoma-frontends.akkoma-fe</li>
    <li>alice-lg</li>
    <li>amarok (amarok-kf5)</li>
    <li>amarok.doc (amarok-kf5.doc)</li>
    <li>anime-downloader</li>
    <li>anime-downloader.dist</li>
    <li>anki</li>
    <li>anki.dist</li>
    <li>anki.doc</li>
    <li>anki.man</li>
    <li>ansible-language-server</li>
    <li>antennas</li>
    <li>antora</li>
    <li>apache-airflow</li>
    <li>apache-airflow.dist</li>
    <li>ariang</li>
    <li>arianna</li>
    <li>assemblyscript</li>
    <li>audiobookshelf</li>
    <li>authelia</li>
    <li>aw-server-rust</li>
    <li>balanceofsatoshis</li>
    <li>base16-builder</li>
    <li>bat-extras.prettybat</li>
    <li>betterbird-unwrapped</li>
    <li>betterbird-unwrapped.debug</li>
    <li>bibtex-tidy</li>
    <li>binaryen</li>
    <li>bitwarden</li>
    <li>bitwarden-cli</li>
    <li>breath-theme</li>
    <li>breitbandmessung</li>
    <li>calamares</li>
    <li>calamares-nixos</li>
    <li>calibre</li>
    <li>cantor (libsForQt5.cantor ,plasma5Packages.cantor)</li>
    <li>castnow</li>
    <li>cdxgen</li>
    <li>changedetection-io</li>
    <li>changedetection-io.dist</li>
    <li>chromium</li>
    <li>chromium.sandbox</li>
    <li>cinnamon.xreader</li>
    <li>cjdns</li>
    <li>cjdns-tools</li>
    <li>clipgrab</li>
    <li>cockpit</li>
    <li>commitizen</li>
    <li>commitizen.dist</li>
    <li>commitlint</li>
    <li>concurrently</li>
    <li>create-cycle-app</li>
    <li>csound-qt</li>
    <li>csslint</li>
    <li>cudaPackages.nsight_compute</li>
    <li>cutemarked-ng</li>
    <li>cutter</li>
    <li>cwltool</li>
    <li>cwltool.dist</li>
    <li>cz-cli</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-voice-note</li>
    <li>deltachat-desktop</li>
    <li>dgraph</li>
    <li>digikam</li>
    <li>djot-js</li>
    <li>dmlive</li>
    <li>docker-compose-language-service</li>
    <li>dot-language-server</li>
    <li>drawio</li>
    <li>drawio-headless</li>
    <li>eagle</li>
    <li>electron-fiddle</li>
    <li>element-desktop</li>
    <li>element-desktop-wayland</li>
    <li>element-web (element-web-unwrapped)</li>
    <li>elmPackages.create-elm-app</li>
    <li>elmPackages.elm</li>
    <li>elmPackages.elm-analyse</li>
    <li>elmPackages.elm-coverage</li>
    <li>elmPackages.elm-doc-preview</li>
    <li>elmPackages.elm-git-install</li>
    <li>elmPackages.elm-language-server</li>
    <li>elmPackages.elm-live</li>
    <li>elmPackages.elm-optimize-level-2</li>
    <li>elmPackages.elm-pages</li>
    <li>elmPackages.elm-review</li>
    <li>elmPackages.elm-spa</li>
    <li>elmPackages.elm-test</li>
    <li>elmPackages.elm-upgrade</li>
    <li>elmPackages.elm-verify-examples</li>
    <li>elmPackages.elm-xref</li>
    <li>elmPackages.elm.doc</li>
    <li>nodejs (elmPackages.nodejs ,nodejs_18)</li>
    <li>nodejs.libv8 (elmPackages.nodejs.libv8 ,nodejs_18.libv8)</li>
    <li>emacsPackages.lsp-bridge</li>
    <li>emanote</li>
    <li>emanote.data</li>
    <li>emojione</li>
    <li>emscripten</li>
    <li>epgstation</li>
    <li>eslint_d</li>
    <li>ethercalc</li>
    <li>evcc</li>
    <li>fast-cli</li>
    <li>faust (faust2)</li>
    <li>faust2alqt</li>
    <li>faust2alsa</li>
    <li>faust2csound</li>
    <li>faust2firefox</li>
    <li>faust2jack</li>
    <li>faust2jackrust</li>
    <li>faust2jaqt</li>
    <li>faust2ladspa</li>
    <li>faust2lv2</li>
    <li>faust2sc</li>
    <li>faustPhysicalModeling</li>
    <li>faustlive</li>
    <li>fcitx5-chinese-addons</li>
    <li>filebrowser</li>
    <li>firefox-beta-unwrapped (firefoxPackages.firefox-beta)</li>
    <li>firefox-beta-unwrapped.debug (firefoxPackages.firefox-beta.debug)</li>
    <li>firefox-beta-unwrapped.symbols (firefoxPackages.firefox-beta.symbols)</li>
    <li>firefox-devedition-unwrapped (firefoxPackages.firefox-devedition)</li>
    <li>firefox-devedition-unwrapped.debug (firefoxPackages.firefox-devedition.debug)</li>
    <li>firefox-devedition-unwrapped.symbols (firefoxPackages.firefox-devedition.symbols)</li>
    <li>firefox-esr-unwrapped (firefox-esr-102-unwrapped ,firefoxPackages.firefox-esr-102)</li>
    <li>firefox-esr-unwrapped.debug (firefox-esr-102-unwrapped.debug ,firefoxPackages.firefox-esr-102.debug)</li>
    <li>firefox-esr-unwrapped.symbols (firefox-esr-102-unwrapped.symbols ,firefoxPackages.firefox-esr-102.symbols)</li>
    <li>xulrunner (firefox-unwrapped)</li>
    <li>xulrunner.debug (firefox-unwrapped.debug)</li>
    <li>xulrunner.symbols (firefox-unwrapped.symbols)</li>
    <li>fixup_yarn_lock</li>
    <li>flexoptix-app</li>
    <li>flood</li>
    <li>forgejo</li>
    <li>forgejo.data</li>
    <li>foxdot (python310Packages.foxdot)</li>
    <li>foxdot.dist (python310Packages.foxdot.dist)</li>
    <li>freecad</li>
    <li>frescobaldi</li>
    <li>frescobaldi.dist</li>
    <li>fx_cast_bridge</li>
    <li>github-copilot-cli</li>
    <li>gitlab</li>
    <li>gitlab-ee</li>
    <li>gitqlient</li>
    <li>globalprotect-openconnect</li>
    <li>gnomeExtensions.pop-shell</li>
    <li>golden-cheetah</li>
    <li>google-clasp</li>
    <li>gotify-server</li>
    <li>gpsbabel-gui</li>
    <li>gpsbabel-gui.doc</li>
    <li>grafana-image-renderer</li>
    <li>graphite-cli</li>
    <li>gtop</li>
    <li>guitarix</li>
    <li>haskellPackages.aeson-typescript</li>
    <li>haskellPackages.aeson-typescript.doc</li>
    <li>haskellPackages.emanote</li>
    <li>haskellPackages.emanote.data</li>
    <li>haskellPackages.emanote.doc</li>
    <li>haskellPackages.servant-typescript</li>
    <li>haskellPackages.servant-typescript.doc</li>
    <li>haskellPackages.tailwind</li>
    <li>haskellPackages.tailwind.doc</li>
    <li>haste-server</li>
    <li>hedgedoc</li>
    <li>heroic</li>
    <li>heroic-unwrapped</li>
    <li>heroku</li>
    <li>hqplayer-desktop</li>
    <li>hqplayer-desktop.doc</li>
    <li>hred</li>
    <li>hueadm</li>
    <li>huggle</li>
    <li>husky</li>
    <li>image_optim</li>
    <li>imapnotify</li>
    <li>imgbrd-grabber</li>
    <li>iosevka</li>
    <li>iosevka-comfy.comfy</li>
    <li>iosevka-comfy.comfy-duo</li>
    <li>iosevka-comfy.comfy-fixed</li>
    <li>iosevka-comfy.comfy-motion</li>
    <li>iosevka-comfy.comfy-motion-duo</li>
    <li>iosevka-comfy.comfy-motion-fixed</li>
    <li>iosevka-comfy.comfy-wide</li>
    <li>iosevka-comfy.comfy-wide-duo</li>
    <li>iosevka-comfy.comfy-wide-fixed</li>
    <li>iosevka-comfy.comfy-wide-motion</li>
    <li>iosevka-comfy.comfy-wide-motion-duo</li>
    <li>iosevka-comfy.comfy-wide-motion-fixed</li>
    <li>isso</li>
    <li>isso.dist</li>
    <li>isso.doc</li>
    <li>jami</li>
    <li>jellyfin</li>
    <li>jellyfin-media-player</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>jellyfin-web</li>
    <li>jellyseerr</li>
    <li>joplin</li>
    <li>kapitonov-plugins-pack</li>
    <li>karma</li>
    <li>kaufkauflist</li>
    <li>kavita</li>
    <li>kbibtex</li>
    <li>kchmviewer</li>
    <li>kdeltachat</li>
    <li>kiwix</li>
    <li>kmymoney</li>
    <li>kuro</li>
    <li>labplot</li>
    <li>latte-dock</li>
    <li>lemmy-ui</li>
    <li>lessc</li>
    <li>lgogdownloader</li>
    <li>librewolf-unwrapped</li>
    <li>librewolf-unwrapped.debug</li>
    <li>libsForQt5.akonadi-calendar (plasma5Packages.akonadi-calendar)</li>
    <li>libsForQt5.akonadi-calendar-tools (plasma5Packages.akonadi-calendar-tools)</li>
    <li>libsForQt5.akonadi-calendar-tools.dev (plasma5Packages.akonadi-calendar-tools.dev)</li>
    <li>libsForQt5.akonadi-calendar.dev (plasma5Packages.akonadi-calendar.dev)</li>
    <li>libsForQt5.akonadi-contacts (plasma5Packages.akonadi-contacts)</li>
    <li>libsForQt5.akonadi-contacts.dev (plasma5Packages.akonadi-contacts.dev)</li>
    <li>libsForQt5.akonadi-import-wizard (plasma5Packages.akonadi-import-wizard)</li>
    <li>libsForQt5.akonadi-import-wizard.dev (plasma5Packages.akonadi-import-wizard.dev)</li>
    <li>libsForQt5.akonadiconsole (plasma5Packages.akonadiconsole)</li>
    <li>libsForQt5.akregator (plasma5Packages.akregator)</li>
    <li>libsForQt5.akregator.dev (plasma5Packages.akregator.dev)</li>
    <li>libsForQt5.alkimia (plasma5Packages.alkimia)</li>
    <li>libsForQt5.angelfish (plasma5Packages.angelfish)</li>
    <li>libsForQt5.aura-browser (plasma5Packages.aura-browser)</li>
    <li>libsForQt5.bismuth (plasma5Packages.bismuth)</li>
    <li>libsForQt5.calendarsupport (plasma5Packages.calendarsupport)</li>
    <li>libsForQt5.calendarsupport.dev (plasma5Packages.calendarsupport.dev)</li>
    <li>libsForQt5.discover (plasma5Packages.discover)</li>
    <li>libsForQt5.eventviews (plasma5Packages.eventviews)</li>
    <li>libsForQt5.eventviews.dev (plasma5Packages.eventviews.dev)</li>
    <li>libsForQt5.falkon (plasma5Packages.falkon)</li>
    <li>libsForQt5.ghostwriter (plasma5Packages.ghostwriter)</li>
    <li>libsForQt5.incidenceeditor (plasma5Packages.incidenceeditor)</li>
    <li>libsForQt5.incidenceeditor.dev (plasma5Packages.incidenceeditor.dev)</li>
    <li>libsForQt5.kaccounts-providers (plasma5Packages.kaccounts-providers)</li>
    <li>libsForQt5.kaddressbook (plasma5Packages.kaddressbook)</li>
    <li>libsForQt5.kalarm (plasma5Packages.kalarm)</li>
    <li>libsForQt5.kalendar (plasma5Packages.kalendar)</li>
    <li>libsForQt5.kde-cli-tools (plasma5Packages.kde-cli-tools)</li>
    <li>libsForQt5.kdepim-addons (plasma5Packages.kdepim-addons)</li>
    <li>libsForQt5.kdepim-runtime (plasma5Packages.kdepim-runtime)</li>
    <li>libsForQt5.kdeplasma-addons (plasma5Packages.kdeplasma-addons)</li>
    <li>libsForQt5.kdev-php (plasma5Packages.kdev-php)</li>
    <li>libsForQt5.kdev-python (plasma5Packages.kdev-python)</li>
    <li>libsForQt5.kdevelop (plasma5Packages.kdevelop)</li>
    <li>libsForQt5.kdevelop-unwrapped (plasma5Packages.kdevelop-unwrapped)</li>
    <li>libsForQt5.kgpg (plasma5Packages.kgpg)</li>
    <li>libsForQt5.khotkeys (plasma5Packages.khotkeys)</li>
    <li>libsForQt5.khotkeys.bin (plasma5Packages.khotkeys.bin)</li>
    <li>libsForQt5.khotkeys.dev (plasma5Packages.khotkeys.dev)</li>
    <li>libsForQt5.kinfocenter (plasma5Packages.kinfocenter)</li>
    <li>libsForQt5.kio-gdrive (plasma5Packages.kio-gdrive)</li>
    <li>libsForQt5.kmail (plasma5Packages.kmail)</li>
    <li>libsForQt5.kmail-account-wizard (plasma5Packages.kmail-account-wizard)</li>
    <li>libsForQt5.kmail.doc (plasma5Packages.kmail.doc)</li>
    <li>libsForQt5.kmailtransport (plasma5Packages.kmailtransport)</li>
    <li>libsForQt5.kmailtransport.dev (plasma5Packages.kmailtransport.dev)</li>
    <li>libsForQt5.kmenuedit (plasma5Packages.kmenuedit)</li>
    <li>libsForQt5.knotes (plasma5Packages.knotes)</li>
    <li>libsForQt5.konqueror (plasma5Packages.konqueror)</li>
    <li>libsForQt5.kontact (plasma5Packages.kontact)</li>
    <li>libsForQt5.korganizer (plasma5Packages.korganizer)</li>
    <li>libsForQt5.krohnkite (plasma5Packages.krohnkite)</li>
    <li>libsForQt5.ksystemstats (plasma5Packages.ksystemstats)</li>
    <li>libsForQt5.kwin-dynamic-workspaces (plasma5Packages.kwin-dynamic-workspaces)</li>
    <li>libsForQt5.kwin-tiling (plasma5Packages.kwin-tiling)</li>
    <li>libsForQt5.kzones (plasma5Packages.kzones)</li>
    <li>libsForQt5.libgravatar (plasma5Packages.libgravatar)</li>
    <li>libsForQt5.libgravatar.dev (plasma5Packages.libgravatar.dev)</li>
    <li>libsForQt5.libkdepim (plasma5Packages.libkdepim)</li>
    <li>libsForQt5.libkgapi (plasma5Packages.libkgapi)</li>
    <li>libsForQt5.libksieve (plasma5Packages.libksieve)</li>
    <li>libsForQt5.libksieve.dev (plasma5Packages.libksieve.dev)</li>
    <li>libsForQt5.libksysguard (plasma5Packages.libksysguard)</li>
    <li>libsForQt5.libksysguard.bin (plasma5Packages.libksysguard.bin)</li>
    <li>libsForQt5.libksysguard.dev (plasma5Packages.libksysguard.dev)</li>
    <li>libsForQt5.mailcommon (plasma5Packages.mailcommon)</li>
    <li>libsForQt5.mailcommon.dev (plasma5Packages.mailcommon.dev)</li>
    <li>libsForQt5.mailimporter (plasma5Packages.mailimporter)</li>
    <li>libsForQt5.marble (plasma5Packages.marble)</li>
    <li>libsForQt5.marble.dev (plasma5Packages.marble.dev)</li>
    <li>libsForQt5.mauikit-calendar (plasma5Packages.mauikit-calendar)</li>
    <li>libsForQt5.mbox-importer (plasma5Packages.mbox-importer)</li>
    <li>libsForQt5.messagelib (plasma5Packages.messagelib)</li>
    <li>libsForQt5.messagelib.dev (plasma5Packages.messagelib.dev)</li>
    <li>libsForQt5.parachute (plasma5Packages.parachute)</li>
    <li>libsForQt5.pim-data-exporter (plasma5Packages.pim-data-exporter)</li>
    <li>libsForQt5.pim-sieve-editor (plasma5Packages.pim-sieve-editor)</li>
    <li>libsForQt5.pimcommon (plasma5Packages.pimcommon)</li>
    <li>libsForQt5.pimcommon.dev (plasma5Packages.pimcommon.dev)</li>
    <li>libsForQt5.plasma-bigscreen (plasma5Packages.plasma-bigscreen)</li>
    <li>libsForQt5.plasma-browser-integration (plasma5Packages.plasma-browser-integration)</li>
    <li>libsForQt5.plasma-desktop (plasma5Packages.plasma-desktop)</li>
    <li>libsForQt5.plasma-disks (plasma5Packages.plasma-disks)</li>
    <li>libsForQt5.plasma-disks.bin (plasma5Packages.plasma-disks.bin)</li>
    <li>libsForQt5.plasma-disks.dev (plasma5Packages.plasma-disks.dev)</li>
    <li>libsForQt5.plasma-mobile (plasma5Packages.plasma-mobile)</li>
    <li>libsForQt5.plasma-remotecontrollers (plasma5Packages.plasma-remotecontrollers)</li>
    <li>libsForQt5.plasma-systemmonitor (plasma5Packages.plasma-systemmonitor)</li>
    <li>libsForQt5.plasma-vault (plasma5Packages.plasma-vault)</li>
    <li>libsForQt5.plasma-workspace (plasma5Packages.plasma-workspace)</li>
    <li>libsForQt5.plasma-workspace.dev (plasma5Packages.plasma-workspace.dev)</li>
    <li>libsForQt5.powerdevil (plasma5Packages.powerdevil)</li>
    <li>qt5.qtwebengine (libsForQt5.qtwebengine)</li>
    <li>qt5.qtwebengine.bin (libsForQt5.qtwebengine.bin)</li>
    <li>qt5.qtwebengine.dev (libsForQt5.qtwebengine.dev)</li>
    <li>qt5.qtwebview (libsForQt5.qtwebview)</li>
    <li>qt5.qtwebview.bin (libsForQt5.qtwebview.bin)</li>
    <li>qt5.qtwebview.dev (libsForQt5.qtwebview.dev)</li>
    <li>libsForQt5.systemsettings (plasma5Packages.systemsettings)</li>
    <li>libsForQt5.systemsettings.bin (plasma5Packages.systemsettings.bin)</li>
    <li>libsForQt5.systemsettings.dev (plasma5Packages.systemsettings.dev)</li>
    <li>libsForQt5.tokodon (plasma5Packages.tokodon)</li>
    <li>libsForQt5.xdg-desktop-portal-kde (plasma5Packages.xdg-desktop-portal-kde)</li>
    <li>libsForQt5.zanshin (plasma5Packages.zanshin)</li>
    <li>lighthouse</li>
    <li>lightly-boehs</li>
    <li>lightly-qt</li>
    <li>ligo</li>
    <li>listmonk</li>
    <li>lldap</li>
    <li>luminanceHDR</li>
    <li>lv_img_conv</li>
    <li>magnetophonDSP.CharacterCompressor</li>
    <li>magnetophonDSP.CompBus</li>
    <li>magnetophonDSP.LazyLimiter</li>
    <li>magnetophonDSP.MBdistortion</li>
    <li>magnetophonDSP.RhythmDelay</li>
    <li>magnetophonDSP.VoiceOfFaust</li>
    <li>magnetophonDSP.faustCompressors</li>
    <li>magnetophonDSP.pluginUtils</li>
    <li>magnetophonDSP.shelfMultiBand</li>
    <li>malt</li>
    <li>mastodon</li>
    <li>mastodon-bot</li>
    <li>matrix-alertmanager</li>
    <li>matrix-appservice-discord</li>
    <li>matrix-appservice-irc</li>
    <li>matrix-appservice-slack</li>
    <li>matrix-hookshot</li>
    <li>matrix-sdk-crypto-nodejs</li>
    <li>matrix-sdk-crypto-nodejs-0_1_0-beta_3</li>
    <li>maui-shell</li>
    <li>mbqn</li>
    <li>mellowplayer</li>
    <li>memos</li>
    <li>merkaartor</li>
    <li>meshcentral</li>
    <li>micropad</li>
    <li>mindforger</li>
    <li>mirakurun</li>
    <li>mjolnir</li>
    <li>mkchromecast</li>
    <li>mkchromecast.dist</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mongosh</li>
    <li>mooSpace</li>
    <li>morgen</li>
    <li>mpvScripts.webtorrent-mpv-hook</li>
    <li>musescore</li>
    <li>muzika</li>
    <li>n8n</li>
    <li>napari (python310Packages.napari)</li>
    <li>napari.dist (python310Packages.napari.dist)</li>
    <li>napi-rs-cli</li>
    <li>natron</li>
    <li>navidrome</li>
    <li>nest-cli</li>
    <li>netlify-cli</li>
    <li>newman</li>
    <li>nextcloud-client</li>
    <li>nextcloud-client.dev</li>
    <li>nmapsi4</li>
    <li>node-manta</li>
    <li>node2nix</li>
    <li>nodejs-slim (nodejs-slim_18)</li>
    <li>nodejs-slim.libv8 (nodejs-slim_18.libv8)</li>
    <li>nodejs-slim_20 (nodejs-slim_latest)</li>
    <li>nodejs-slim_20.libv8 (nodejs-slim_latest.libv8)</li>
    <li>nodejs_20 (nodejs_latest)</li>
    <li>nodejs_20.libv8 (nodejs_latest.libv8)</li>
    <li>notepadqq</li>
    <li>ntfy-sh</li>
    <li>onionshare-gui</li>
    <li>onionshare-gui.dist</li>
    <li>onlykey</li>
    <li>open-music-kontrollers.mephisto</li>
    <li>open-stage-control</li>
    <li>openboard</li>
    <li>openmoji-black</li>
    <li>openmoji-color</li>
    <li>openshot-qt</li>
    <li>openshot-qt.dist</li>
    <li>osmtogeojson</li>
    <li>outline</li>
    <li>p4v</li>
    <li>pairdrop</li>
    <li>pandoc-drawio-filter</li>
    <li>pandoc-drawio-filter.dist</li>
    <li>paperless-ngx</li>
    <li>patray</li>
    <li>patray.dist</li>
    <li>peertube</li>
    <li>pentobi</li>
    <li>pgadmin4</li>
    <li>pgadmin4-desktopmode</li>
    <li>pgadmin4-desktopmode.dist</li>
    <li>pgadmin4.dist</li>
    <li>photofield</li>
    <li>photoprism</li>
    <li>plausible</li>
    <li>playwright (python310Packages.playwright)</li>
    <li>playwright-driver</li>
    <li>playwright-test</li>
    <li>playwright.dist (python310Packages.playwright.dist)</li>
    <li>plex-media-player</li>
    <li>pm2</li>
    <li>pnpm-lock-export</li>
    <li>podman-desktop</li>
    <li>polaris</li>
    <li>polaris-web</li>
    <li>polychromatic</li>
    <li>pomerium</li>
    <li>postcss-cli</li>
    <li>pot</li>
    <li>powerdns-admin</li>
    <li>pre-commit</li>
    <li>pre-commit.dist</li>
    <li>prefetch-yarn-deps</li>
    <li>pscid</li>
    <li>psi</li>
    <li>psitransfer</li>
    <li>pulp</li>
    <li>pulsar</li>
    <li>puppeteer-cli</li>
    <li>purescript-psa</li>
    <li>pyCA</li>
    <li>pyCA.dist</li>
    <li>pyright</li>
    <li>python-qt</li>
    <li>python310Packages.bokeh</li>
    <li>python310Packages.bokeh.dist</li>
    <li>python310Packages.datashader</li>
    <li>python310Packages.datashader.dist</li>
    <li>python310Packages.django-hijack</li>
    <li>python310Packages.django-hijack.dist</li>
    <li>python310Packages.django-js-reverse</li>
    <li>python310Packages.django-js-reverse.dist</li>
    <li>python310Packages.holoviews</li>
    <li>python310Packages.holoviews.dist</li>
    <li>python310Packages.hvplot</li>
    <li>python310Packages.hvplot.dist</li>
    <li>python310Packages.intake</li>
    <li>python310Packages.intake.dist</li>
    <li>python310Packages.livelossplot</li>
    <li>python310Packages.livelossplot.dist</li>
    <li>python310Packages.magicgui</li>
    <li>python310Packages.magicgui.dist</li>
    <li>python310Packages.mkdocs-drawio-exporter</li>
    <li>python310Packages.mkdocs-drawio-exporter.dist</li>
    <li>python310Packages.napari-npe2</li>
    <li>python310Packages.napari-npe2.dist</li>
    <li>python310Packages.nbdev</li>
    <li>python310Packages.nbdev.dist</li>
    <li>python310Packages.nbsmoke</li>
    <li>python310Packages.nbsmoke.dist</li>
    <li>python310Packages.panel</li>
    <li>python310Packages.panel.dist</li>
    <li>python310Packages.pscript</li>
    <li>python310Packages.pscript.dist</li>
    <li>python310Packages.pyqt6-webengine</li>
    <li>python310Packages.pyqt6-webengine.dev</li>
    <li>python310Packages.pyqt6-webengine.dist</li>
    <li>python310Packages.pyqtwebengine</li>
    <li>python310Packages.pyqtwebengine.dev</li>
    <li>python310Packages.pyqtwebengine.dist</li>
    <li>python310Packages.pyside2</li>
    <li>python310Packages.pyside2-tools</li>
    <li>python310Packages.pyside6</li>
    <li>python310Packages.pytest-playwright</li>
    <li>python310Packages.pytest-playwright.dist</li>
    <li>python310Packages.pywebview</li>
    <li>python310Packages.pywebview.dist</li>
    <li>spyder (python310Packages.spyder)</li>
    <li>spyder.dist (python310Packages.spyder.dist)</li>
    <li>python310Packages.timetagger</li>
    <li>python310Packages.timetagger.dist</li>
    <li>python311Packages.bokeh</li>
    <li>python311Packages.bokeh.dist</li>
    <li>python311Packages.django-hijack</li>
    <li>python311Packages.django-hijack.dist</li>
    <li>python311Packages.django-js-reverse</li>
    <li>python311Packages.django-js-reverse.dist</li>
    <li>python311Packages.foxdot</li>
    <li>python311Packages.foxdot.dist</li>
    <li>python311Packages.holoviews</li>
    <li>python311Packages.holoviews.dist</li>
    <li>python311Packages.hvplot</li>
    <li>python311Packages.hvplot.dist</li>
    <li>python311Packages.intake</li>
    <li>python311Packages.intake.dist</li>
    <li>python311Packages.livelossplot</li>
    <li>python311Packages.livelossplot.dist</li>
    <li>python311Packages.mkdocs-drawio-exporter</li>
    <li>python311Packages.mkdocs-drawio-exporter.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>python311Packages.nbsmoke</li>
    <li>python311Packages.nbsmoke.dist</li>
    <li>python311Packages.panel</li>
    <li>python311Packages.panel.dist</li>
    <li>python311Packages.playwright</li>
    <li>python311Packages.playwright.dist</li>
    <li>python311Packages.pscript</li>
    <li>python311Packages.pscript.dist</li>
    <li>python311Packages.pyqt6-webengine</li>
    <li>python311Packages.pyqt6-webengine.dev</li>
    <li>python311Packages.pyqt6-webengine.dist</li>
    <li>python311Packages.pyqtwebengine</li>
    <li>python311Packages.pyqtwebengine.dev</li>
    <li>python311Packages.pyqtwebengine.dist</li>
    <li>python311Packages.pyside6</li>
    <li>python311Packages.pytest-playwright</li>
    <li>python311Packages.pytest-playwright.dist</li>
    <li>python311Packages.pywebview</li>
    <li>python311Packages.pywebview.dist</li>
    <li>python311Packages.spyder</li>
    <li>python311Packages.spyder.dist</li>
    <li>python311Packages.timetagger</li>
    <li>python311Packages.timetagger.dist</li>
    <li>qmapshack</li>
    <li>qolibri</li>
    <li>qsyncthingtray</li>
    <li>qt6.full (qt6Packages.full)</li>
    <li>qt6.qtwebengine (qt6Packages.qtwebengine)</li>
    <li>qt6.qtwebengine.dev (qt6Packages.qtwebengine.dev)</li>
    <li>qt6.qtwebview (qt6Packages.qtwebview)</li>
    <li>qt6.qtwebview.dev (qt6Packages.qtwebview.dev)</li>
    <li>qtcreator-qt6</li>
    <li>quarto</li>
    <li>quartoMinimal</li>
    <li>qutebrowser</li>
    <li>qutebrowser-qt6</li>
    <li>qutebrowser-qt6.dist</li>
    <li>qutebrowser.dist</li>
    <li>r2modman</li>
    <li>ratman</li>
    <li>readability-cli</li>
    <li>redoc-cli</li>
    <li>renderdoc</li>
    <li>resumed</li>
    <li>retext</li>
    <li>retext.dist</li>
    <li>reveal-md</li>
    <li>rmfakecloud</li>
    <li>rssguard</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>sapling</li>
    <li>sapling.dist</li>
    <li>schildichat-desktop</li>
    <li>schildichat-desktop-wayland</li>
    <li>schildichat-web</li>
    <li>sdrangel</li>
    <li>seafile-client</li>
    <li>semantic-release</li>
    <li>semantik</li>
    <li>sharedown</li>
    <li>sharing</li>
    <li>shepherd</li>
    <li>shot-scraper</li>
    <li>shot-scraper.dist</li>
    <li>shout</li>
    <li>sigil</li>
    <li>skrooge</li>
    <li>sl1-to-photon</li>
    <li>slack</li>
    <li>slimerjs</li>
    <li>sloc</li>
    <li>slskd</li>
    <li>sourcehut.buildsrht</li>
    <li>sourcehut.buildsrht.dist</li>
    <li>sourcehut.coresrht</li>
    <li>sourcehut.coresrht.dist</li>
    <li>sourcehut.gitsrht</li>
    <li>sourcehut.gitsrht.dist</li>
    <li>sourcehut.hgsrht</li>
    <li>sourcehut.hgsrht.dist</li>
    <li>sourcehut.hubsrht</li>
    <li>sourcehut.hubsrht.dist</li>
    <li>sourcehut.listssrht</li>
    <li>sourcehut.listssrht.dist</li>
    <li>sourcehut.mansrht</li>
    <li>sourcehut.mansrht.dist</li>
    <li>sourcehut.metasrht</li>
    <li>sourcehut.metasrht.dist</li>
    <li>sourcehut.pastesrht</li>
    <li>sourcehut.pastesrht.dist</li>
    <li>sourcehut.todosrht</li>
    <li>sourcehut.todosrht.dist</li>
    <li>stellarium</li>
    <li>streamdeck-ui</li>
    <li>streamdeck-ui.dist</li>
    <li>stremio</li>
    <li>styx</li>
    <li>styx.lib</li>
    <li>styx.themes</li>
    <li>subsurface</li>
    <li>sunshine</li>
    <li>supercollider</li>
    <li>supercollider-with-plugins</li>
    <li>supercollider-with-sc3-plugins</li>
    <li>supercolliderPlugins.sc3-plugins</li>
    <li>supercollider_scel</li>
    <li>swagger-cli</li>
    <li>synapse-admin</li>
    <li>syncplay</li>
    <li>syncthingtray</li>
    <li>tageditor</li>
    <li>tambura</li>
    <li>tandoor-recipes</li>
    <li>teams</li>
    <li>teams-for-linux</li>
    <li>teamspeak_client</li>
    <li>teamviewer</li>
    <li>teamviewer.dev</li>
    <li>teck-programmer</li>
    <li>teleport (teleport_12)</li>
    <li>teleport.client (teleport_12.client)</li>
    <li>teleport_11</li>
    <li>teleport_11.client</li>
    <li>tellico</li>
    <li>terminal-stocks</li>
    <li>texmaker</li>
    <li>theLoungePlugins.plugins.closepms</li>
    <li>theLoungePlugins.plugins.giphy</li>
    <li>theLoungePlugins.plugins.shortcuts</li>
    <li>theLoungePlugins.themes.abyss</li>
    <li>theLoungePlugins.themes.amoled</li>
    <li>theLoungePlugins.themes.amoled-sourcecodepro</li>
    <li>theLoungePlugins.themes.bdefault</li>
    <li>theLoungePlugins.themes.bmorning</li>
    <li>theLoungePlugins.themes.chord</li>
    <li>theLoungePlugins.themes.classic</li>
    <li>theLoungePlugins.themes.common</li>
    <li>theLoungePlugins.themes.crypto</li>
    <li>theLoungePlugins.themes.discordapp</li>
    <li>theLoungePlugins.themes.dracula</li>
    <li>theLoungePlugins.themes.dracula-official</li>
    <li>theLoungePlugins.themes.flat-blue</li>
    <li>theLoungePlugins.themes.flat-dark</li>
    <li>theLoungePlugins.themes.gruvbox</li>
    <li>theLoungePlugins.themes.hexified</li>
    <li>theLoungePlugins.themes.ion</li>
    <li>theLoungePlugins.themes.light</li>
    <li>theLoungePlugins.themes.midnight</li>
    <li>theLoungePlugins.themes.mininapse</li>
    <li>theLoungePlugins.themes.monokai-console</li>
    <li>theLoungePlugins.themes.mortified</li>
    <li>theLoungePlugins.themes.neuron-fork</li>
    <li>theLoungePlugins.themes.new-morning</li>
    <li>theLoungePlugins.themes.new-morning-compact</li>
    <li>theLoungePlugins.themes.nologo</li>
    <li>theLoungePlugins.themes.nord</li>
    <li>theLoungePlugins.themes.onedark</li>
    <li>theLoungePlugins.themes.purplenight</li>
    <li>theLoungePlugins.themes.scoutlink</li>
    <li>theLoungePlugins.themes.seraphimrp</li>
    <li>theLoungePlugins.themes.solarized</li>
    <li>theLoungePlugins.themes.solarized-fork-monospace</li>
    <li>theLoungePlugins.themes.zenburn</li>
    <li>theLoungePlugins.themes.zenburn-monospace</li>
    <li>theLoungePlugins.themes.zenburn-sourcecodepro</li>
    <li>thelounge</li>
    <li>thunderbird-unwrapped (thunderbirdPackages.thunderbird ,thunderbirdPackages.thunderbird-102)</li>
    <li>thunderbird-unwrapped.debug (thunderbirdPackages.thunderbird-102.debug ,thunderbirdPackages.thunderbird.debug)</li>
    <li>thunderbird-unwrapped.symbols (thunderbirdPackages.thunderbird-102.symbols ,thunderbirdPackages.thunderbird.symbols)</li>
    <li>tinygo</li>
    <li>torq</li>
    <li>triton</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>turbo</li>
    <li>twilio-cli</li>
    <li>twspace-crawler</li>
    <li>typescript</li>
    <li>uivonim</li>
    <li>ungoogled-chromium</li>
    <li>ungoogled-chromium.sandbox</li>
    <li>uppy-companion</li>
    <li>uptime-kuma</li>
    <li>v2raya</li>
    <li>vencord</li>
    <li>vencord-web-extension</li>
    <li>vhs</li>
    <li>vieb</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vimPlugins.cmp-copilot</li>
    <li>vimPlugins.cmp-npm</li>
    <li>vimPlugins.coc-clangd</li>
    <li>vimPlugins.coc-cmake</li>
    <li>vimPlugins.coc-css</li>
    <li>vimPlugins.coc-diagnostic</li>
    <li>vimPlugins.coc-docker</li>
    <li>vimPlugins.coc-emmet</li>
    <li>vimPlugins.coc-eslint</li>
    <li>vimPlugins.coc-explorer</li>
    <li>vimPlugins.coc-flutter</li>
    <li>vimPlugins.coc-git</li>
    <li>vimPlugins.coc-go</li>
    <li>vimPlugins.coc-haxe</li>
    <li>vimPlugins.coc-highlight</li>
    <li>vimPlugins.coc-html</li>
    <li>vimPlugins.coc-java</li>
    <li>vimPlugins.coc-jest</li>
    <li>vimPlugins.coc-json</li>
    <li>vimPlugins.coc-lists</li>
    <li>vimPlugins.coc-ltex</li>
    <li>vimPlugins.coc-markdownlint</li>
    <li>vimPlugins.coc-metals</li>
    <li>vimPlugins.coc-nginx</li>
    <li>vimPlugins.coc-pairs</li>
    <li>vimPlugins.coc-prettier</li>
    <li>vimPlugins.coc-pyright</li>
    <li>vimPlugins.coc-python</li>
    <li>vimPlugins.coc-r-lsp</li>
    <li>vimPlugins.coc-rls</li>
    <li>vimPlugins.coc-rust-analyzer</li>
    <li>vimPlugins.coc-sh</li>
    <li>vimPlugins.coc-smartf</li>
    <li>vimPlugins.coc-snippets</li>
    <li>vimPlugins.coc-solargraph</li>
    <li>vimPlugins.coc-spell-checker</li>
    <li>vimPlugins.coc-sqlfluff</li>
    <li>vimPlugins.coc-stylelint</li>
    <li>vimPlugins.coc-sumneko-lua</li>
    <li>vimPlugins.coc-tabnine</li>
    <li>vimPlugins.coc-texlab</li>
    <li>vimPlugins.coc-toml</li>
    <li>vimPlugins.coc-tslint</li>
    <li>vimPlugins.coc-tslint-plugin</li>
    <li>vimPlugins.coc-tsserver</li>
    <li>vimPlugins.coc-ultisnips</li>
    <li>vimPlugins.coc-vetur</li>
    <li>vimPlugins.coc-vimlsp</li>
    <li>vimPlugins.coc-vimtex</li>
    <li>vimPlugins.coc-wxml</li>
    <li>vimPlugins.coc-yaml</li>
    <li>vimPlugins.coc-yank</li>
    <li>vimPlugins.copilot-vim</li>
    <li>vimPlugins.markdown-preview-nvim</li>
    <li>vimPlugins.nvim-treesitter-parsers.d</li>
    <li>vimPlugins.nvim-treesitter-parsers.mlir</li>
    <li>vimPlugins.nvim-treesitter-parsers.ocamllex</li>
    <li>vimPlugins.nvim-treesitter-parsers.swift</li>
    <li>vimPlugins.nvim-treesitter-parsers.teal</li>
    <li>vivisect</li>
    <li>vivisect.dist</li>
    <li>vlang</li>
    <li>vnote</li>
    <li>vsce</li>
    <li>vscode</li>
    <li>vscode-extensions.ms-python.vscode-pylance</li>
    <li>vscode-extensions.rust-lang.rust-analyzer</li>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
    <li>vscode-fhs</li>
    <li>vscode-langservers-extracted</li>
    <li>vscode-with-extensions</li>
    <li>vscodium</li>
    <li>vscodium-fhs</li>
    <li>wacomtablet</li>
    <li>wails</li>
    <li>wasm-bindgen-cli</li>
    <li>wasm-strip</li>
    <li>wasm-text-gen</li>
    <li>wast-refmt</li>
    <li>wayback</li>
    <li>webassemblyjs-cli</li>
    <li>webassemblyjs-repl</li>
    <li>webcord</li>
    <li>webcord-vencord</li>
    <li>webmacs</li>
    <li>webmacs.dist</li>
    <li>webtorrent_desktop</li>
    <li>weylus</li>
    <li>whitebophir</li>
    <li>windmill</li>
    <li>woodpecker-server</li>
    <li>wrangler</li>
    <li>wring</li>
    <li>xplorer</li>
    <li>yacas-gui</li>
    <li>yaml-language-server</li>
    <li>yarn</li>
    <li>yarn-lock-converter</li>
    <li>yarn2nix</li>
    <li>ycmd</li>
    <li>yuzu-early-access</li>
    <li>yuzu-mainline</li>
    <li>zammad</li>
    <li>zeal (zeal-qt5)</li>
    <li>zeal-qt6</li>
    <li>zigbee2mqtt</li>
    <li>zincsearch</li>
    <li>zrythm</li>
    <li>zx</li>
  </ul>
</details>

lilyinstarlight commented on pkgs/development/web/nodejs/v20.nix:
```diff
@@ -16,5 +16,6 @@ buildNodejs {
     ./disable-darwin-v8-system-instrumentation-node19.patch
     ./bypass-darwin-xcrun-node16.patch
     ./node-npm-build-npm-package-logic.patch
+    ./common-gypi-cross.patch
```
This patch is not nixpkgs-specific, right? Do you think you could make an upstream PR for this patch and `fetchpatch` it here?

I'm sure we wouldn't be the only ones to benefit from nodejs having working aarch64-linux -> x86_64-linux cross :)

Thank you for this work!

tie committed:
nodejs: use fetchpatch for cross-compilation patches

lilyinstarlight commented on pkgs/development/web/nodejs/common-gypi-cross.patch:
```diff
@@ -0,0 +1,84 @@
+Fixes target toolchain arguments being passed to the host toolchain when
```
Can you remove this file now, since it is fetched with `fetchpatch`?

lilyinstarlight reviewed (commented):
I left one comment, but otherwise this looks fine enough for me now. Do you have any other thoughts @marsam?

lilyinstarlight commented on pkgs/development/web/nodejs/nodejs.nix:
```diff
@@ -50,9 +52,10 @@ let
+    depsBuildBuild = [ buildPackages.stdenv.cc openssl libuv zlib icu ];
 
     buildInputs = lib.optionals stdenv.isDarwin [ CoreServices ApplicationServices ]
+      ++ lib.optional isCross runtimeShellPackage # for patchShebangs
```
Actually one more comment too: is this just meant to be `bash` and if it is needed at runtime, shouldn't it just be unconditionally added instead of only added for cross?

tie replied:
Yes, bash/sh. I think it should be possible to remove bash/sh dependency since all scripts seem to be simple wrappers that invoke node on the corresponding JS script. I’ve tried that, but IIRC some other packages use node-gyp shell script wrapper and I didn’t investigate failures.

<details>
<summary><code>rg --sort path /nix/store</code></summary>

```
include/node/config.gypi
2:{ 'target_defaults': {'cflags': [], 'default_configuration': 'Release', 'defines': ['NODE_OPENSSL_CONF_NAME=nodejs_conf', 'ICU_NO_USER_DATA_OVERRIDE'], 'include_dirs': ['/nix/store/v1ivffsk8kcnzmkanf1hdw8jdhf8cnni-zlib-1.2.13-dev/include', '/nix/store/16f23j7k72rdjji6g0106h2ss1ghx3ys-libuv-1.45.0-dev/include', '/nix/store/xsc0lsv634lv5zk9dfdfaysv3x79y52r-openssl-3.0.9-dev/include', '/nix/store/viagaap65dxbijv0pahim3dkzkakhlpa-icu4c-73.1-dev/include'], 'libraries': ['-L/nix/store/lxgxpd3yl9y8mwkqs905lms4vzwca3hy-zlib-1.2.13/lib', '-lz', '-L/nix/store/0m9z7g2f5y42w19jxykw9rhmaqg25arj-libuv-1.45.0/lib', '-luv', '-L/nix/store/xim52l8jwiasmmalsnwcy51f3l77cggm-openssl-3.0.9/lib', '-lcrypto', '-lssl', '-L/nix/store/h79hac9j683jcm3q0fky6kbj4flnivbh-icu4c-73.1/lib', '-licui18n', '-licuuc', '-licudata']},
337:                 'node_prefix': '/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0',

lib/node_modules/corepack/dist/corepack.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/npm.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/npx.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/pnpm.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/pnpx.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/yarn.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/dist/yarnpkg.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/corepack/shims/corepack
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/npm
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/npx
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/pnpm
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/pnpx
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/yarn
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/corepack/shims/yarnpkg
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/bin/node-gyp-bin/node-gyp
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/bin/npm
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/bash

lib/node_modules/npm/bin/npm-cli.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/bin/npx
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/bash

lib/node_modules/npm/bin/npx-cli.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/lib/utils/completion.sh
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/bash

lib/node_modules/npm/node_modules/@npmcli/arborist/bin/index.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/@npmcli/run-script/lib/node-gyp-bin/node-gyp
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/node_modules/color-support/bin.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/cssesc/bin/cssesc
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/mkdirp/bin/cmd.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/node-gyp/gyp/gyp
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/node_modules/node-gyp/gyp/tools/emacs/run-unit-tests.sh
1:#!/nix/store/vaibdw2zi029qw0mqxcbhg1zhzbvwxcv-bash-5.2-p15/bin/sh

lib/node_modules/npm/node_modules/node-gyp/node_modules/nopt/bin/nopt.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/node-gyp/node_modules/which/bin/node-which
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/nopt/bin/nopt.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/pacote/lib/bin.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/qrcode-terminal/bin/qrcode-terminal.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/rimraf/bin.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/semver/bin/semver.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/sigstore/bin/sigstore.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node

lib/node_modules/npm/node_modules/which/bin/which.js
1:#!/nix/store/yfhrwc6ycmw0p3cvqiafa07hiwhk5r34-nodejs-18.16.0/bin/node
```
</details>

Also `HOST_PATH` in stdenv.mkDerivation is empty when cross-compiling (without `buildInputs`) but contains implicit coreutils, bash and other stuff when `hostPlatform == buildPlatform`. Not sure if this is intended behavior 🤷

lilyinstarlight replied:
A bash dependency is fine, I'm just meaning why would it only be relevant for cross?

If it just pulling in the implicit bash from `nativeBuildInputs` for non-cross since `strictDeps == false`, we should be uncoditionally adding it rather than relying on implicit dependencies

tie replied:
Agreed, should I also set `strictDeps = true`?

SuperSandro2000 replied:
Please make it unconditional so that we notice early if cross breaks in an update

tie replied:
Done 8aaec56348450d639db2a700b178f5715b993976

tie committed:
nodejs: set strictDeps and patch bash shebangs

tie commented:
@SuperSandro2000 @lilyinstarlight, gentle ping, is there anything else blocking this PR?

marsam reviewed (approved):
LGTM. Thank you!

SuperSandro2000 commented:
I am not that fit in cross to know if this is the correct way or not.

marsam commented:
I think the patch `common-gypi-cross.patch` is already included in v20.5.0
@GrahamcOfBorg eval

tie committed:
nodejs_20: remove upstreamed patch

tie commented:
>@marsam I think the patch common-gypi-cross.patch is already included in v20.5.0

You are right, I’ve rebased the branch and removed the patch for v20.
